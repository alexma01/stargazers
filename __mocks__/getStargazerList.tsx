import {Stargazer} from '../src/models/Stargazer';

export const stargazersMocks: Stargazer[] = [
  {
    username: 'test1',
    avatar: 'https://avatars.githubusercontent.com/u/18049606?v=4',
  },
  {
    username: 'test2',
    avatar: 'https://avatars.githubusercontent.com/u/18049606?v=4',
  },
  {
    username: 'test3',
    avatar: 'https://avatars.githubusercontent.com/u/18049606?v=4',
  },
  {
    username: 'test4',
    avatar: '',
  },
];
