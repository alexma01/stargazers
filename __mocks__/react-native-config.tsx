export default {
  GITHUB_TOKEN: '',
  GITHUB_ENDPOINT: 'https://api.github.com/',
};
