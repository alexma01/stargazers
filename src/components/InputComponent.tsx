import React from 'react';
import {StyleSheet, Text, TextInput, View} from 'react-native';

interface Props {
  testID?: string;
  placeholder?: string;
  text: string;
  onChangeText?: (text: string) => void;
}

export const InputComponent = ({
  placeholder,
  onChangeText,
  text,
  testID,
}: Props) => {
  return (
    <View>
      <Text style={styles.label}>{placeholder}</Text>
      <TextInput
        testID={testID}
        placeholder={placeholder}
        style={styles.input}
        onChangeText={onChangeText}
        value={text}
        autoCapitalize="none"
      />
    </View>
  );
};

const styles = StyleSheet.create({
  label: {
    color: 'black',
    fontWeight: 'bold',
    fontSize: 15,
    marginLeft: 15,
  },
  input: {
    margin: 15,
    height: 40,
    borderColor: 'gray',
    borderWidth: 1,
    color: 'gray',
    borderBottomWidth: 2,
    backgroundColor: 'white',
    paddingLeft: 15,
    borderRadius: 10,
  },
});
