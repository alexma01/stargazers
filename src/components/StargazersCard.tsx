import React, {memo} from 'react';
import {Image, StyleSheet, Text, View} from 'react-native';
import {Stargazer} from '../models/Stargazer';

interface Props {
  stargazer: Stargazer;
}

export const StargazersCard = memo(
  (props: Props) => {
    return (
      <View style={styles.container}>
        <Image style={styles.avatar} source={{uri: props.stargazer.avatar}} />
        <Text style={styles.textStyle}>{props.stargazer.username}</Text>
      </View>
    );
  },
  (prev, next) => {
    return prev.stargazer.username === next.stargazer.username;
  },
);

const styles = StyleSheet.create({
  container: {
    backgroundColor: 'white',
    width: '100%',
    flexDirection: 'row',
    padding: '1%',
    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.25,
    shadowRadius: 3.84,
    elevation: 5,
    borderRadius: 10,
    alignItems: 'center',
    marginTop: 5,
  },
  avatar: {
    width: 70,
    height: 70,
  },
  textStyle: {
    marginLeft: 10,
  },
});
