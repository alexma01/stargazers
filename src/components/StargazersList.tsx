import React from 'react';
import {
  FlatList,
  View,
  StyleSheet,
  Text,
  ActivityIndicator,
  ListRenderItemInfo,
} from 'react-native';
import {Stargazer} from '../models/Stargazer';
import {StargazersCard} from './StargazersCard';

interface Props {
  stargazers: Stargazer[];
  loading: boolean;
  onEndReached?: () => void;
}

export const StargazersList = (props: Props) => {
  return (
    <View style={styles.container}>
      {props.stargazers.length > 0 && (
        <FlatList
          testID="StargazersList"
          contentContainerStyle={{marginHorizontal: 10}}
          style={styles.list}
          data={props.stargazers}
          renderItem={({item}: ListRenderItemInfo<Stargazer>) => (
            <StargazersCard stargazer={item} />
          )}
          keyExtractor={(item: Stargazer) => item.username}
          ItemSeparatorComponent={() => <View style={{height: 10}} />}
          onEndReached={props.onEndReached}
          onEndReachedThreshold={0.1}
        />
      )}
      {props.stargazers.length === 0 && (
        <View style={styles.no_results}>
          <Text>NO RESULTS</Text>
        </View>
      )}
      {props.loading && (
        <View style={styles.loadingContainer}>
          <ActivityIndicator size={'large'} />
        </View>
      )}
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  list: {
    flex: 1,
  },
  no_results: {
    height: '100%',
    justifyContent: 'center',
    alignContent: 'center',
    alignItems: 'center',
  },
  loadingContainer: {
    justifyContent: 'center',
    alignContent: 'center',
    alignItems: 'center',
    textAlignVertical: 'center',
  },
});
