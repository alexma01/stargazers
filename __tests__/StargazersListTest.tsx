import TestRenderer from 'react-test-renderer';
import React from 'react';
import {StargazersList} from '../src/components/StargazersList';
import {stargazersMocks} from '../__mocks__/getStargazerList';

describe('StargazersList', () => {
  it('should match the snapshot with loading true and stargazers list', () => {
    expect(
      TestRenderer.create(
        <StargazersList loading={true} stargazers={stargazersMocks} />,
      ),
    ).toMatchSnapshot();
  });

  it('should match the snapshot with loading false and stargazers list', () => {
    expect(
      TestRenderer.create(
        <StargazersList loading={false} stargazers={stargazersMocks} />,
      ),
    ).toMatchSnapshot();
  });

  it('should match the snapshot with loading true and empty stargazers list', () => {
    expect(
      TestRenderer.create(<StargazersList loading={true} stargazers={[]} />),
    ).toMatchSnapshot();
  });

  it('should match the snapshot with loading false and empty stargazers list', () => {
    expect(
      TestRenderer.create(<StargazersList loading={false} stargazers={[]} />),
    ).toMatchSnapshot();
  });
});
