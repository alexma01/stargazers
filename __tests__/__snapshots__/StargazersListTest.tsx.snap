// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StargazersList should match the snapshot with loading false and empty stargazers list 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    style={
      {
        "alignContent": "center",
        "alignItems": "center",
        "height": "100%",
        "justifyContent": "center",
      }
    }
  >
    <Text>
      NO RESULTS
    </Text>
  </View>
</View>
`;

exports[`StargazersList should match the snapshot with loading false and stargazers list 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
>
  <RCTScrollView
    ItemSeparatorComponent={[Function]}
    contentContainerStyle={
      {
        "marginHorizontal": 10,
      }
    }
    data={
      [
        {
          "avatar": "https://avatars.githubusercontent.com/u/18049606?v=4",
          "username": "test1",
        },
        {
          "avatar": "https://avatars.githubusercontent.com/u/18049606?v=4",
          "username": "test2",
        },
        {
          "avatar": "https://avatars.githubusercontent.com/u/18049606?v=4",
          "username": "test3",
        },
        {
          "avatar": "",
          "username": "test4",
        },
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={0.1}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={[]}
    style={
      {
        "flex": 1,
      }
    }
    testID="StargazersList"
    viewabilityConfigCallbackPairs={[]}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            {
              "alignItems": "center",
              "backgroundColor": "white",
              "borderRadius": 10,
              "elevation": 5,
              "flexDirection": "row",
              "marginTop": 5,
              "padding": "1%",
              "shadowColor": "#000",
              "shadowOffset": {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.25,
              "shadowRadius": 3.84,
              "width": "100%",
            }
          }
        >
          <Image
            source={
              {
                "uri": "https://avatars.githubusercontent.com/u/18049606?v=4",
              }
            }
            style={
              {
                "height": 70,
                "width": 70,
              }
            }
          />
          <Text
            style={
              {
                "marginLeft": 10,
              }
            }
          >
            test1
          </Text>
        </View>
        <View
          style={
            {
              "height": 10,
            }
          }
        />
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            {
              "alignItems": "center",
              "backgroundColor": "white",
              "borderRadius": 10,
              "elevation": 5,
              "flexDirection": "row",
              "marginTop": 5,
              "padding": "1%",
              "shadowColor": "#000",
              "shadowOffset": {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.25,
              "shadowRadius": 3.84,
              "width": "100%",
            }
          }
        >
          <Image
            source={
              {
                "uri": "https://avatars.githubusercontent.com/u/18049606?v=4",
              }
            }
            style={
              {
                "height": 70,
                "width": 70,
              }
            }
          />
          <Text
            style={
              {
                "marginLeft": 10,
              }
            }
          >
            test2
          </Text>
        </View>
        <View
          style={
            {
              "height": 10,
            }
          }
        />
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            {
              "alignItems": "center",
              "backgroundColor": "white",
              "borderRadius": 10,
              "elevation": 5,
              "flexDirection": "row",
              "marginTop": 5,
              "padding": "1%",
              "shadowColor": "#000",
              "shadowOffset": {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.25,
              "shadowRadius": 3.84,
              "width": "100%",
            }
          }
        >
          <Image
            source={
              {
                "uri": "https://avatars.githubusercontent.com/u/18049606?v=4",
              }
            }
            style={
              {
                "height": 70,
                "width": 70,
              }
            }
          />
          <Text
            style={
              {
                "marginLeft": 10,
              }
            }
          >
            test3
          </Text>
        </View>
        <View
          style={
            {
              "height": 10,
            }
          }
        />
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            {
              "alignItems": "center",
              "backgroundColor": "white",
              "borderRadius": 10,
              "elevation": 5,
              "flexDirection": "row",
              "marginTop": 5,
              "padding": "1%",
              "shadowColor": "#000",
              "shadowOffset": {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.25,
              "shadowRadius": 3.84,
              "width": "100%",
            }
          }
        >
          <Image
            source={
              {
                "uri": "",
              }
            }
            style={
              {
                "height": 70,
                "width": 70,
              }
            }
          />
          <Text
            style={
              {
                "marginLeft": 10,
              }
            }
          >
            test4
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;

exports[`StargazersList should match the snapshot with loading true and empty stargazers list 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    style={
      {
        "alignContent": "center",
        "alignItems": "center",
        "height": "100%",
        "justifyContent": "center",
      }
    }
  >
    <Text>
      NO RESULTS
    </Text>
  </View>
  <View
    style={
      {
        "alignContent": "center",
        "alignItems": "center",
        "justifyContent": "center",
        "textAlignVertical": "center",
      }
    }
  >
    <ActivityIndicator
      size="large"
    />
  </View>
</View>
`;

exports[`StargazersList should match the snapshot with loading true and stargazers list 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
>
  <RCTScrollView
    ItemSeparatorComponent={[Function]}
    contentContainerStyle={
      {
        "marginHorizontal": 10,
      }
    }
    data={
      [
        {
          "avatar": "https://avatars.githubusercontent.com/u/18049606?v=4",
          "username": "test1",
        },
        {
          "avatar": "https://avatars.githubusercontent.com/u/18049606?v=4",
          "username": "test2",
        },
        {
          "avatar": "https://avatars.githubusercontent.com/u/18049606?v=4",
          "username": "test3",
        },
        {
          "avatar": "",
          "username": "test4",
        },
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={0.1}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={[]}
    style={
      {
        "flex": 1,
      }
    }
    testID="StargazersList"
    viewabilityConfigCallbackPairs={[]}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            {
              "alignItems": "center",
              "backgroundColor": "white",
              "borderRadius": 10,
              "elevation": 5,
              "flexDirection": "row",
              "marginTop": 5,
              "padding": "1%",
              "shadowColor": "#000",
              "shadowOffset": {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.25,
              "shadowRadius": 3.84,
              "width": "100%",
            }
          }
        >
          <Image
            source={
              {
                "uri": "https://avatars.githubusercontent.com/u/18049606?v=4",
              }
            }
            style={
              {
                "height": 70,
                "width": 70,
              }
            }
          />
          <Text
            style={
              {
                "marginLeft": 10,
              }
            }
          >
            test1
          </Text>
        </View>
        <View
          style={
            {
              "height": 10,
            }
          }
        />
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            {
              "alignItems": "center",
              "backgroundColor": "white",
              "borderRadius": 10,
              "elevation": 5,
              "flexDirection": "row",
              "marginTop": 5,
              "padding": "1%",
              "shadowColor": "#000",
              "shadowOffset": {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.25,
              "shadowRadius": 3.84,
              "width": "100%",
            }
          }
        >
          <Image
            source={
              {
                "uri": "https://avatars.githubusercontent.com/u/18049606?v=4",
              }
            }
            style={
              {
                "height": 70,
                "width": 70,
              }
            }
          />
          <Text
            style={
              {
                "marginLeft": 10,
              }
            }
          >
            test2
          </Text>
        </View>
        <View
          style={
            {
              "height": 10,
            }
          }
        />
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            {
              "alignItems": "center",
              "backgroundColor": "white",
              "borderRadius": 10,
              "elevation": 5,
              "flexDirection": "row",
              "marginTop": 5,
              "padding": "1%",
              "shadowColor": "#000",
              "shadowOffset": {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.25,
              "shadowRadius": 3.84,
              "width": "100%",
            }
          }
        >
          <Image
            source={
              {
                "uri": "https://avatars.githubusercontent.com/u/18049606?v=4",
              }
            }
            style={
              {
                "height": 70,
                "width": 70,
              }
            }
          />
          <Text
            style={
              {
                "marginLeft": 10,
              }
            }
          >
            test3
          </Text>
        </View>
        <View
          style={
            {
              "height": 10,
            }
          }
        />
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            {
              "alignItems": "center",
              "backgroundColor": "white",
              "borderRadius": 10,
              "elevation": 5,
              "flexDirection": "row",
              "marginTop": 5,
              "padding": "1%",
              "shadowColor": "#000",
              "shadowOffset": {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.25,
              "shadowRadius": 3.84,
              "width": "100%",
            }
          }
        >
          <Image
            source={
              {
                "uri": "",
              }
            }
            style={
              {
                "height": 70,
                "width": 70,
              }
            }
          />
          <Text
            style={
              {
                "marginLeft": 10,
              }
            }
          >
            test4
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <View
    style={
      {
        "alignContent": "center",
        "alignItems": "center",
        "justifyContent": "center",
        "textAlignVertical": "center",
      }
    }
  >
    <ActivityIndicator
      size="large"
    />
  </View>
</View>
`;
